### ISSUE TITLE
pygame.Font rendered, anti-aliased text with transparent background looks worse/faded in pygame 2

### ISSUE THREAD

**Issue by MyreMylar** [pygame/pygame#1413]
A subtle issue I've been noticing, particularly when working with small font sizes. Something in the way anti-aliased text with a transparent background is rendered has changed between pygame 1.9.6 and pygame 2, and not for the better in my opinion. This is visible on Windows, no idea if it is also present on other platforms at the moment.

I've personally seen the issue in dev3 and dev4 and can investigate different versions further if needed.

Here's how some text looks under pygame 1.9.6:
![pygame_1_9](https://user-images.githubusercontent.com/13382426/66736994-a55c3500-ee62-11e9-8a56-a83bedecf6aa.png)

And here's the same text under pygame 2 dev4:
![pygame_2](https://user-images.githubusercontent.com/13382426/66737026-c15fd680-ee62-11e9-8b02-a63d2614b49c.png)

The font used here is Fira Code, but I don't think that particularly matters. 

If you render text with a background colour set then everything looks fine/consistent across versions, but that is obviously not ideal in all cases as text background colours have no alpha at all. Even if they did, in the example above when I want to stick text on to a semi-transparent background the text background would look odd blitted onto an already semi-transparent rectangle.

This issue is more noticeable with dark text on a light background.

**Comment by Pololot64** [comment]
Could this also have to do with the alpha issue? #1289

**Comment by nthykier** [comment]
Hi @MyreMylar 

Would you happen to have a small test case that reproduces the issue for you that we can use as a baseline for testing this issue?

**Comment by MyreMylar** [comment]
I can probably work one up. Give me a bit.

**Comment by MyreMylar** [comment]
I suspect it's probably easier to see the way the alpha changes can smudge letters together and obscure fine details with a serif font, but I've just used the default pygame font here for ease of running:

    import pygame


    pygame.init()

    window_surface = pygame.display.set_mode((640, 480))

    background_1 = pygame.Surface((320, 480))
    background_1.fill(pygame.Color(20,170,20))
    background_2 = pygame.Surface((320, 480))
    background_2.fill(pygame.Color(170,20,20))

    intermediate_alpha_surface = pygame.Surface((640, 480), flags=pygame.SRCALPHA)
    intermediate_alpha_surface.fill(pygame.Color(255, 255, 255, 160))

    font = pygame.font.Font(None, 14)

    font_surface = font.render('Text with alpha background', True, pygame.Color(0,0,0,255))
    font_rect = font_surface.get_rect()
    font_rect.center = (320, 240)
    intermediate_alpha_surface.blit(font_surface, font_rect)

    running = True

    while running:
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                running = False

        window_surface.blit(background_1, (0, 0))
        window_surface.blit(background_2, (320, 0))
        window_surface.blit(intermediate_alpha_surface, (0, 0))
        pygame.display.update()

Here's how it looks in **1.9.6**:
![pygame_1](https://user-images.githubusercontent.com/13382426/79020252-44db5580-7b70-11ea-9670-ca77612df7ad.png)

And here's how it looks in **2.0.0.dev6**:
![pygame_2](https://user-images.githubusercontent.com/13382426/79020263-4efd5400-7b70-11ea-80ae-85f0f28a7ba9.png)
(You can see the 'k' and the 'g' here appearing to sort of mush into each other due to the loss of alpha detail and the distance opening up between the 'o' and 'u' )

And zoomed in -

**1.9.6**:
![image](https://user-images.githubusercontent.com/13382426/79020569-11e59180-7b71-11ea-8566-f0fc9950bca5.png)

**2.0.0.dev6**:
![image](https://user-images.githubusercontent.com/13382426/79020525-f5495980-7b70-11ea-85b6-8096d65807df.png)

**Comment by nthykier** [comment]
@MyreMylar Thanks for that. :)

Interestingly, I cannot reproduce it with master@e07618a05545.  I am hoping this means it is fixed but it could also be an OS related difference between the two of us.

For reference, I updated the snippet to include the pygame snippet to ensure my screenshots were of the right test. :)

```python
import pygame


pygame.init()

window_surface = pygame.display.set_mode((640, 480))

background_1 = pygame.Surface((320, 480))
background_1.fill(pygame.Color(20,170,20))
background_2 = pygame.Surface((320, 480))
background_2.fill(pygame.Color(170,20,20))

intermediate_alpha_surface = pygame.Surface((640, 480), flags=pygame.SRCALPHA)
intermediate_alpha_surface.fill(pygame.Color(255, 255, 255, 160))

font = pygame.font.Font(None, 14)

font_surface = font.render('Text with alpha background', True, pygame.Color(0,0,0,255))
font_rect = font_surface.get_rect()
font_rect.center = (320, 240)
intermediate_alpha_surface.blit(font_surface, font_rect)

font_pygame_version = font.render(pygame.version.ver, True, pygame.Color(0,0,0,255))
font_pygame_version_rect = font_pygame_version.get_rect()
font_pygame_version_rect.center = (320, 275)

intermediate_alpha_surface.blit(font_pygame_version, font_pygame_version_rect)

running = True

while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    window_surface.blit(background_1, (0, 0))
    window_surface.blit(background_2, (320, 0))
    window_surface.blit(intermediate_alpha_surface, (0, 0))
    pygame.display.update()
```

**Comment by nthykier** [comment]
What I see (pygame 1.9.6/SDL1.2 - distro build):

![gh-1413-py38-pg-196](https://user-images.githubusercontent.com/429402/79040145-eb0d7680-7bd5-11ea-8181-19262d23f8d3.png)

What I see (pygame 2/SDL2 - self-comile):

![gh-1413-py38-pg-2-master-e07618a05545](https://user-images.githubusercontent.com/429402/79040147-ed6fd080-7bd5-11ea-8973-cbc4086501ed.png)

**Comment by nthykier** [comment]
I just tried with 2.0.0.dev6 (e9de5c27d6a7887cea08f37e0ad48d2d20f5328e) and I cannot reproduce it there either.  So I am suspecting that there is another difference triggering the bug for you but not for me.

**Comment by MightyJosip** [comment]
This one is interesting. What I found is that if you install pygame with SDL_ttf version 2.0.14 I get this bug, but if I install SDL_ttf version 2.0.15 I don't get this bug. Also looking at patch notes for SDL_ttf I found out this change in ttf " * Text rendering functions now use the alpha component of the text colors". So this could be the reason it works now

**Comment by nthykier** [comment]
@MightyJosip Thanks for the hint.

@MyreMylar Can you confirm MightyJosip's finding? :)

For reference,  I am using SDL_tff 2.0.11.

**Comment by MightyJosip** [comment]
2.0.11 is the last one for SDL 1. Could be that they failed something when they wrote the code for SDL 2 in 2.0.12, and then they fixed it in 2.0.15

**Comment by MyreMylar** [comment]
> @MyreMylar Can you confirm MightyJosip's finding? :)

Sadly, I haven't got as far as building pygame from source yet (it's on my Todo list) so I can't confirm that for you, or at least not anytime soon. Right now I'm just using the windows build uploaded to pypi for both 1.9.6 and 2.0.0.dev6. 

Sounds promising though!

**Comment by MightyJosip** [comment]
This has been fixed in #1641 and confirmed by @MyreMylar on discord. Thanks to everyone who participated in fixing this critical bug.

**Comment by MyreMylar** [comment]
I have _actually_ tested it now (this morning), not sure what was misinterpreted on discord, and can confirm that the text from the example now looks the same between 1.9.8 and 2.0.0.dev7. This definitely seems fixed.
